testing/test_mark.py

```diff
@@ -22,6 +22,19 @@ def test_pytest_mark_notcallable(self):
         mark = Mark()
         pytest.raises((AttributeError, TypeError), mark)
 
+    def test_mark_with_param(self):
+        def some_function(abc):
+            pass
+
+        class SomeClass(object):
+            pass
+
+        assert pytest.mark.fun(some_function) is some_function
+        assert pytest.mark.fun.with_args(some_function) is not some_function
```

Comment:
I might be misunderstanding the change then, because IIUC `pytest.mark.fun.with_args(some_function)` is a new feature (so no other tests use it yet) and we expect it to return a `MarkInfo`, but we are not making that assertion anywhere.

But no biggie really, if you think this is fine no worries.